### ISSUE TITLE
Prefab->SetRoot broken

### ISSUE THREAD

**Issue by honzapatCZ** [FlaxEngine/FlaxEngine#489]
This feature doesnt work at all. References are lost, hierarchy is not as it should, just broken. You are far better of creating that prefab you want with BreakPrefabLink and then recreate it.

**Comment by stefnotch** [comment]
I actually managed to reproduce something here 🥳 

Steps to reproduce one of the setroot bugs
1. Create a prefab, add an empty actor and an animated model.
2. Save the prefab

![image](https://user-images.githubusercontent.com/10220080/116192794-9e3c0580-a72e-11eb-8856-0b3dfbdea100.png)


3. Randomly click on one of the actors, set root and occasionally save the prefab

![image](https://user-images.githubusercontent.com/10220080/116192889-cc214a00-a72e-11eb-84c5-f7386383b450.png)

4. At some point, it'll look like this (see right side)

![image](https://user-images.githubusercontent.com/10220080/116192774-97ad8e00-a72e-11eb-8888-e364506832e7.png)

5. At that point, the output log will also show a warning message

```
[ 00:03:16.577 ]: [Info] Updating referencing prefabs
[ 00:03:16.577 ]: [Info] Prefab updated! 0 ms
[ 00:03:17.053 ]: [Info] Saving assets cache to 'C:\Users/Stefnotch/Documents/GitHub/Flax/anime-rendering/Cache/AssetsCache.dat', entries: 172
[ 00:03:17.183 ]: [Warning] Failed to setup values and UI for item SkinnedModel
[ 00:03:17.183 ]: [Warning] Exception has been thrown by the target of an invocation.
[ 00:03:17.183 ]: [Warning]   at System.Reflection.MonoProperty.GetValue (System.Object obj, System.Object[] index) [0x00081] in <ccba8f4931be433d979e22082f9d6862>:0 
  at FlaxEditor.Scripting.ScriptMemberInfo.GetValue (System.Object obj) [0x00015] in C:\Users\Stefnotch\Documents\GitHub\Flax\FlaxEngine\Source\Editor\Scripting\ScriptType.cs:638 
  at FlaxEditor.CustomEditors.ValueContainer..ctor (FlaxEditor.Scripting.ScriptMemberInfo info, FlaxEditor.CustomEditors.ValueContainer instanceValues) [0x000d1] in C:\Users\Stefnotch\Documents\GitHub\Flax\FlaxEngine\Source\Editor\CustomEditors\Values\ValueContainer.cs:258 
  at FlaxEditor.CustomEditors.Editors.GenericEditor+ItemInfo.GetValues (FlaxEditor.CustomEditors.ValueContainer instanceValues) [0x00001] in C:\Users\Stefnotch\Documents\GitHub\Flax\FlaxEngine\Source\Editor\CustomEditors\Editors\GenericEditor.cs:151 
  at FlaxEditor.CustomEditors.Editors.GenericEditor.Initialize (FlaxEditor.CustomEditors.LayoutElementsContainer layout) [0x002db] in C:\Users\Stefnotch\Documents\GitHub\Flax\FlaxEngine\Source\Editor\CustomEditors\Editors\GenericEditor.cs:534 
[ 00:03:17.418 ]: [Info] Creating package at 'C:\Users/Stefnotch/Documents/GitHub/Flax/anime-rendering/Cache/Thumbnails/cache_9be9ebebee2643cfbd2b88df4b8fa261.flax'. Silent Mode: tru
```

**Comment by honzapatCZ** [comment]
Ye that's one of the versions, other is,
it will crash
it will look like it works but actually after you close the prefab and open it back up it will be as before, but the old root will be duplicated.

**Comment by mafiesto4** [comment]
I fixed this feature and made it more reliable 7fc3e65849bfa9bb88510b831ee4ba8bf0e30c15. Reopen this ticket if occurs again (on latest master or in the next engine update `1.2`).
